core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -385,8 +385,6 @@ private[spark] object ResourceUtils extends Logging {
     val pluginClasses = sparkConf.get(RESOURCES_DISCOVERY_PLUGIN) :+ discoveryScriptPlugin
     val resourcePlugins = Utils.loadExtensions(classOf[ResourceDiscoveryPlugin], pluginClasses,
       sparkConf)
-    // apply each plugin until one of them returns the information for this resource
-    var riOption: Optional[ResourceInformation] = Optional.empty()
```

Comment:
SPARK-30689 add `riOption` but it was not used because there is a definition with the same name in `resourcePlugins.foreach` loop